ui/decredmaterial/container.go

```diff
@@ -0,0 +1,70 @@
+package decredmaterial
+
+import (
+	"gioui.org/layout"
+	"gioui.org/unit"
+)
+
+type Container struct {
+	container                       *layout.List
+	scrollbar                       Scrollbar
+	totalContentHeight              float32
+	hasCalculatedTotalContentHeight bool
+	scrollbarThicknessPercentage    float32
+}
+
+func (t *Theme) Container() *Container {
+	return &Container{
+		container:                       &layout.List{Axis: layout.Vertical},
+		scrollbar:                       t.Scrollbar(new(Float), 0, 100),
+		totalContentHeight:              0,
+		hasCalculatedTotalContentHeight: false,
+		scrollbarThicknessPercentage:    1.3,
+	}
+}
+
+func (c *Container) calculateTotalContentHeight(gtx layout.Context, widgets []func(gtx C) D) {
+	for i := range widgets {
+		index := i
+		(&layout.List{Axis: layout.Vertical}).Layout(gtx, 1, func(gtx C, i int) D {
+			dim := layout.UniformInset(unit.Dp(0)).Layout(gtx, widgets[index])
+			c.totalContentHeight += float32(dim.Size.Y)
+			return layout.Dimensions{}
+		})
+	}
+	c.hasCalculatedTotalContentHeight = true
+}
+
+func (c *Container) Layout(gtx layout.Context, widgets []func(gtx C) D) layout.Dimensions {
+	maxSize := gtx.Constraints.Max
+	c.container.Position.First = int(float32(len(widgets)) * (c.scrollbar.Position() / float32(maxSize.Y)))
+
+	totalVisibleHeight := float32(maxSize.Y)
+	visibleFraction := totalVisibleHeight / c.totalContentHeight
+
+	scrollbarThickness := (c.scrollbarThicknessPercentage / 100) * float32(1)
+
+	return layout.Flex{}.Layout(gtx,
+		layout.Flexed(1-scrollbarThickness, func(gtx C) D {
+			if !c.hasCalculatedTotalContentHeight {
+				c.calculateTotalContentHeight(gtx, widgets)
+				return layout.Dimensions{}
+			}
+
+			return layout.Inset{
+				Right: unit.Dp(15),
+			}.Layout(gtx, func(gtx C) D {
+				return c.container.Layout(gtx, len(widgets), func(gtx C, i int) D {
+					return layout.UniformInset(unit.Dp(0)).Layout(gtx, widgets[i])
+				})
+			})
+		}),
+		layout.Flexed(scrollbarThickness, func(gtx C) D {
+			// don't display scrollbar if total content height is less than or equal to container height
+			if c.totalContentHeight <= totalVisibleHeight {
+				return layout.Dimensions{}
+			}
+			return c.scrollbar.Layout(gtx, visibleFraction)
+		}),
+	)
+}
```

Comment:
this can be in the scrollbar.go file and named "ScrollContainer" instead of just "Container"